src/exodus/test/build_tx_tests.cpp

```diff
@@ -0,0 +1,159 @@
+
+    std::string fromAddress = CBitcoinAddress(pubkey.GetID()).ToString();
+
+    uint256 txid;
+    std::string rawHex;
+    BOOST_CHECK_EQUAL(
+        0, // No error
+        exodus::WalletTxBuilder(fromAddress, "", "", 0, data, txid, rawHex, false)
+    );
+
+    CTransaction decTx;
+    BOOST_CHECK(DecodeHexTx(decTx, rawHex));
+
+    BOOST_CHECK(!decTx.IsSigmaSpend());
+
+    // calculate class c encode
+    std::vector<std::pair<CScript, int64_t>> vecOutputs;
+    Exodus_Encode_ClassC(data, vecOutputs);
+
+    // assert script.
+    BOOST_CHECK_EQUAL(vecOutputs.size() + 1, decTx.vout.size());
+}
+
+BOOST_AUTO_TEST_CASE(wallettxbuilder_create_sigma_without_mints)
+{
+    pwalletMain->SetBroadcastTransactions(true);
+    CreateAndProcessEmptyBlocks(200, scriptPubKey);
+
+    std::vector<unsigned char> data(nMaxDatacarrierBytes + 1, 0x01);
+
+    uint256 txid;
+    std::string rawHex;
+    BOOST_CHECK_EQUAL(
+        MPRPCErrorCode::MP_INPUTS_INVALID,
+        exodus::WalletTxBuilder("", "", "", 0, data, txid, rawHex, false, exodus::InputMode::SIGMA)
+    );
+}
+
+BOOST_AUTO_TEST_CASE(wallettxbuilder_create_sigma_with_toolarge_data)
+{
+    pwalletMain->SetBroadcastTransactions(true);
+    CreateAndProcessEmptyBlocks(200, scriptPubKey);
+
+    string stringError;
+    BOOST_CHECK_MESSAGE(pwalletMain->CreateZerocoinMintModel(
+            stringError, {{"1", 10}}, SIGMA), stringError + " - Create Mint failed");
+
+    CreateAndProcessBlock({}, scriptPubKey);
+    CreateAndProcessEmptyBlocks(5, scriptPubKey);
+
+    std::vector<unsigned char> data(nMaxDatacarrierBytes + 1);
+
+    uint256 txid;
+    std::string rawHex;
+    BOOST_CHECK_EQUAL(
+        MPRPCErrorCode::MP_ENCODING_ERROR,
+        exodus::WalletTxBuilder("", "", "", 0, data, txid, rawHex, false, exodus::InputMode::SIGMA)
+    );
+}
+
+BOOST_AUTO_TEST_CASE(wallettxbuilder_create_sigma_success)
+{
+    pwalletMain->SetBroadcastTransactions(true);
+    CreateAndProcessEmptyBlocks(200, scriptPubKey);
+
+    string stringError;
+    BOOST_CHECK_MESSAGE(pwalletMain->CreateZerocoinMintModel(
+            stringError, {{"1", 10}}, SIGMA), stringError + " - Create Mint failed");
+
+    CreateAndProcessBlock({}, scriptPubKey);
+    CreateAndProcessEmptyBlocks(5, scriptPubKey);
+
+    std::vector<unsigned char> data(80);
+
+    uint256 txid;
+    std::string rawHex;
+    BOOST_CHECK_EQUAL(
+        0, // No error
+        exodus::WalletTxBuilder("", "", "", 0, data, txid, rawHex, false, exodus::InputMode::SIGMA)
+    );
+
+    CTransaction decTx;
+    BOOST_CHECK(DecodeHexTx(decTx, rawHex));
+
+    BOOST_CHECK(decTx.IsSigmaSpend());
+
+    // calculate class c encode
+    std::vector<std::pair<CScript, int64_t>> vecOutputs;
+    Exodus_Encode_ClassC(data, vecOutputs);
+
+    auto opReturnCount = std::count_if(decTx.vout.begin(), decTx.vout.end(), [](CTxOut const &out) {
+        return out.scriptPubKey[0] == OP_RETURN;
+    });
+
+    BOOST_CHECK_EQUAL(vecOutputs.size(), opReturnCount);
```

Comment:
Same as previous issue. Just checking if it is class C and is sigma spend transaction will be more easy to read.